https://bugs.gentoo.org/934785
https://bugs.kde.org/show_bug.cgi?id=489343
https://invent.kde.org/network/kdeconnect-kde/-/merge_requests/732

From 7274c62c7c6418a88bdb0b5ce558d5fd0e749ec8 Mon Sep 17 00:00:00 2001
From: Albert Vaca Cintora <albertvaka@gmail.com>
Date: Thu, 12 Sep 2024 08:03:07 +0000
Subject: [PATCH 1/2] Revert "Revert "Use absolute path in D-Bus service file""

This reverts commit 01a2db20c49729d70e4bdf800bc7e889d254e36e

(cherry picked from commit 0477bc93d01649ed88fbe6738751b800f11df4b6)
---
 daemon/org.kde.kdeconnect.service.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/daemon/org.kde.kdeconnect.service.in b/daemon/org.kde.kdeconnect.service.in
index 95d9a1594..4c972e4d6 100644
--- a/daemon/org.kde.kdeconnect.service.in
+++ b/daemon/org.kde.kdeconnect.service.in
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=org.kde.kdeconnect
-Exec=kdeconnectd
+Exec==@KDE_INSTALL_FULL_BINDIR@/kdeconnectd
-- 
GitLab


From a7045bfe0d2c6521d5fd2806e076eee3ddb8f863 Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Thu, 12 Sep 2024 20:46:35 +0200
Subject: [PATCH 2/2] Don't set an absolute D-Bus service path on Windows and
 macOS

(cherry picked from commit f00e04c07d64641e811ba663bd9fa9c51c7b27aa)
---
 daemon/CMakeLists.txt                | 3 +++
 daemon/org.kde.kdeconnect.service.in | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/daemon/CMakeLists.txt b/daemon/CMakeLists.txt
index 34fb74e70..5f2d0271e 100644
--- a/daemon/CMakeLists.txt
+++ b/daemon/CMakeLists.txt
@@ -20,6 +20,9 @@ ecm_mark_nongui_executable(kdeconnectd)
 
 configure_file(org.kde.kdeconnect.daemon.desktop.cmake ${CMAKE_CURRENT_BINARY_DIR}/org.kde.kdeconnect.daemon.desktop)
 
+if (UNIX AND NOT APPLE)
+    set(kdeconnectd_prefix "${KDE_INSTALL_FULL_BINDIR}/")
+endif()
 configure_file(org.kde.kdeconnect.service.in ${CMAKE_CURRENT_BINARY_DIR}/org.kde.kdeconnect.service)
 
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/org.kde.kdeconnect.daemon.desktop DESTINATION ${KDE_INSTALL_AUTOSTARTDIR})
diff --git a/daemon/org.kde.kdeconnect.service.in b/daemon/org.kde.kdeconnect.service.in
index 4c972e4d6..cb3f5ddbe 100644
--- a/daemon/org.kde.kdeconnect.service.in
+++ b/daemon/org.kde.kdeconnect.service.in
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=org.kde.kdeconnect
-Exec==@KDE_INSTALL_FULL_BINDIR@/kdeconnectd
+Exec=@kdeconnectd_prefix@kdeconnectd
-- 
GitLab
